Applications/Spire/Source/OrderImbalanceIndicator/OrderImbalanceIndicatorTableModel.cpp

```diff
@@ -0,0 +1,155 @@
+#include "Spire/OrderImbalanceIndicator/OrderImbalanceIndicatorTableModel.hpp"
+#include <chrono>
+#include <boost/date_time/gregorian/gregorian.hpp>
+
+using namespace Beam::TimeService;
+using namespace boost;
+using namespace boost::posix_time;
+using namespace boost::signals2;
+using namespace Nexus;
+using namespace Spire;
+
+namespace {
+  const auto EXPIRATION_TIMEOUT_MS = 1000;
+
+  template<typename T>
+  auto get_update(const T& potential,
+      const T& current) -> optional<std::any> {
+    if(potential == current) {
+      return none;
+    }
+    return std::make_any<T>(potential);
+  }
+}
+
+OrderImbalanceIndicatorTableModel::OrderImbalanceIndicatorTableModel(
+    std::shared_ptr<OrderImbalanceIndicatorModel> source,
+    TimeClientBox clock)
+    : m_source(std::move(source)),
+      m_clock(std::move(clock)) {
+  m_expiration_timer.setInterval(EXPIRATION_TIMEOUT_MS);
```

Comment:
My initial idea was to attach a callback for each imbalance in the table, where the callback would be called when that imbalance is considered 'expired' and should be removed from the table. This would ensure that the functionality of the table matches the spec'd behavior where the imbalances are removed as they expire, but I thought this might be excessive if there were hundreds or potentially over a thousand imbalances in the table.

I'll reinvestigate this idea and see if I can come up with an implementation that is more efficient than the existing implementation and doesn't introduce concurrency issues with multiple timers. It may be that a callback-based implementation is faster than the current implementation because each existing imbalance won't have to be iterated over every second.